### PR TITLE
chore: package dependencies upgraded

### PR DESCRIPTION
**Pull request by alex-gilin**

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -18,21 +18,21 @@ deploy-build: &deploy-build
     - run: npm publish --access=public
 
 # ------  build configuration/setup (how?) ------
-version: 2
+version: 2.1 # use CircleCI 2.1
 jobs:
   build-node14:
     docker:
-      - image: circleci/node:14-browsers
+      - image: cimg/node:14.18-browsers
```
Comment (bd82) [inline .circleci/config.yml:25]: why are we using a fixed version of nodejs 14 (14.18) instead of latest? 14.x?

File: .circleci/config.yml [hunk 2]
```diff
@@ -18,21 +18,21 @@ deploy-build: &deploy-build
     - run: npm publish --access=public
 
 # ------  build configuration/setup (how?) ------
-version: 2
+version: 2.1 # use CircleCI 2.1
```
Comment (bd82) [inline .circleci/config.yml:21]: this comment does not add anything

File: package.json [hunk 3]
```diff
@@ -1,101 +1,101 @@
...
-  },
-  "typings": "api.d.ts"
-}
+{
+  "name": "@sap-webide/webide-client-tools",
+  "version": "3.1.3",
+  "license": "Apache-2.0",
+  "main": "src/api.js",
+  "dependencies": {
+    "@types/webpack": "5.28.0",
```
Comment (bd82) [inline package.json:7]: the major version of `@types/webpack` does not seem to match the major version of `webpack` (below)

File: package.json [hunk 4]
```diff
@@ -1,101 +1,101 @@
...
+    "acorn": "8.6.0",
```
Comment (bd82) [inline package.json:8]: where there any breaking changes from acron 7 --> 8?

File: package.json [hunk 5]
```diff
@@ -1,101 +1,101 @@
...
+    "connect": "3.7.0",
+    "fs-extra": "10.0.0",
```
Comment (bd82) [inline package.json:10]: were there any breaking changes from `fs-extra` 4 --> 10? if so what were they?

File: package.json [hunk 6]
```diff
@@ -1,101 +1,101 @@
...
+    "glob": "7.2.0",
+    "http": "0.0.0",
+    "http-proxy-middleware": "0.21.0",
+    "js-string-escape": "1.0.1",
+    "lodash": "4.17.21",
+    "optional-require": "1.1.8",
+    "requirejs": "2.3.6",
+    "serve-static": "1.14.1",
+    "webpack": "4.46.0",
+    "which": "2.0.2"
```
Comment (bd82) [inline package.json:20]: were there breaking changes in `which` from 1.x to 2.x? if so what?

File: package.json [hunk 7]
```diff
@@ -1,101 +1,101 @@
...
+  },
+  "peerDependencies": {
+    "karma": ">= 1.7.0",
+    "karma-chrome-launcher": "3.1.0",
+    "karma-coverage": "2.0.3",
```
Comment (bd82) [inline package.json:25]: there peerDeps changes likely mean this is a major version (breaking) from our perspective
as consumers would have to adapt their package.json files.

**Comment by bd82** [comment]
There are too many changes in this PR. some of which may break the project at runtime (unlikely that  100% of functionality is tested in an automated manner...)

1. `Prettier` version change should be done in a **seprate** PR as it touches dozens of file and makes reviewing the PR diffcult.
2. Changing the TypeDoc output folder should be done ina **separate** PR as once again it touches many files.
   - do the github pages settings need be adjusted after you moved the generated sources?
   - did you test the generated webpage locally?
3. minor/patch version updates should be handled differently than major version changes.
    - Also dev-deps vs deps changes may need different handling.
4. updating CI images should likely be done separatly (if we need to revert it).
5. Updating our peerDeps should likely be considered a breaking change.

In regards to point 3 above:
- patch/minor version changes are generally safe.
- dev-deps updates (even major) are generally safe, as long as we release a new version and test it afterwards
- major changes in Deps section should be done carefully (if at all) as they can include breaking changes.

In regards to our next version number:
- We have to consider if we have any **breaking change** caused by this change, for example if we upgraded a 3rd party 
  library that dropped support for nodejs 12, maybe we should update the engine field in package.json and go up one major version too, (and expose the info on our new breaking change to the consumers).

**Comment by bd82** [comment]
You can merge **seprate** PRs for:
- prettier
- circleci
- typedoc

without waiting for sync review (to reduce ping-pong...).
I will review them async.
Naturally you should test required things post-merge, e.g:
- does the website get updated?
- or maybe the website gets updated only after release?

File: package.json (hunk 3 above)
Comment (alex-gilin) [reply package.json:7]: my understanding, there is no direct correlation (version convention) between versions @types/webpack vs webpack

File: package.json (hunk 4 above)
Comment (alex-gilin) [reply package.json:8]: you probably skipped it in the code https://github.com/SAP/webide-client-tools/blob/15e4fef4dc90cf44152c4e3c59d146c7d7588822/src/bundling.js#L749

File: package.json (hunk 5 above)
Comment (alex-gilin) [reply package.json:10]: example -> https://github.com/SAP/webide-client-tools/blob/15e4fef4dc90cf44152c4e3c59d146c7d7588822/src/bundling.js#L141 - removed unnecessary hack in fs.copySync

File: package.json (hunk 6 above)
Comment (alex-gilin) [reply package.json:20]: they did not meet in our use

File: package.json (hunk 7 above)
Comment (alex-gilin) [reply package.json:25]: np, I'll revert it

**Comment by alex-gilin** [comment]
separated into 4 different PRs

**Pull request closed without merging**